precompiles/utils/src/data/xcm.rs

```diff
@@ -248,3 +250,33 @@ impl EvmData for MultiLocation {
 		EvmDataWriter::write_pointer(writer, inner_writer);
 	}
 }
+
+// For Currencies
+impl EvmData for (Address, U256) {
+	fn read(reader: &mut EvmDataReader, gasometer: &mut Gasometer) -> EvmResult<Self> {
+		let address: Address = reader.read(gasometer)?;
```

Comment:
Should also use a pointer like you did with `MultiAsset`.

Reply:
What I have seen in Remix is that this is only true for structs that have a type with a pointer. The tests that I made seem to assert that

Reply:
E.g., a struct like
```
// A Currency is defined by address and the amount to be transferred
    struct Currency {
        address currency_address;
        uint256 amount;
    }
```
The encoding from Remix seems not to be a pointer, but rather the concatenation of each of the fields

Reply:
Indeed, encoding seems to not use pointer if the size is statically known.